### PR TITLE
Fix DPI-C and protect-lib if port is implicit logic

### PR DESCRIPTION
**Pull request by yTakatsukasa**
I found that if input is an unpacked array of 1 bit wire without `logic` keyword such as `input in[2]`, the generated protect-lib wrapper contains LOGIC_IMPLICIT.
And the temporary variable for such case was svLogicVecVal, but it should be svLogic.

This PR takes care of such case that is tested in 89ceb2e5b19a8627acf2af0752bebb2c0444c392 .

### CODE REVIEW

File: src/V3EmitV.cpp [hunk 1]
```diff
@@ -563,7 +563,11 @@ class EmitVBaseVisitor VL_NOT_FINAL : public EmitCBaseVisitor {
     }
     virtual void visit(AstBasicDType* nodep) override {
         if (nodep->isSigned()) putfs(nodep, "signed ");
-        putfs(nodep, nodep->prettyName());
+        if (nodep->keyword() == AstBasicDTypeKwd::LOGIC_IMPLICIT) {
+            putfs(nodep, "logic");
+        } else {
+            putfs(nodep, nodep->prettyName());
+        }
```
Comment (wsnyder) [inline src/V3EmitV.cpp:570]: I think prettyName for LOGIC_IMPLICIT should be changed to just "logic" as the user doesn't care about LOGIC_IMPLICIT, I'd also add a comment it needs to be logic to appease V3EmitV.

File: src/V3AstNodes.h [hunk 2]
```diff
@@ -924,7 +924,8 @@ class AstBasicDType final : public AstNodeDType {
         return keyword() == AstBasicDTypeKwd::BIT && isRanged();
     }
     bool isDpiLogicVec() const {  // DPI uses svLogicVecVal
-        return keyword().isFourstate() && !(keyword() == AstBasicDTypeKwd::LOGIC && !isRanged());
+        return keyword().isFourstate() && !(keyword() == AstBasicDTypeKwd::LOGIC && !isRanged())
+               && !(keyword() == AstBasicDTypeKwd::LOGIC_IMPLICIT && width1());
```
Comment (wsnyder) [inline src/V3AstNodes.h:928]:
```suggestion
        return keyword().isFourstate() && !((keyword() == AstBasicDTypeKwd::LOGIC
                        || keyword() == AstBasicDTypeKwd::LOGIC_IMPLICIT) && !isRanged());
```
But see intro comment.

**Review by wsnyder — COMMENTED** [review]
I wonder if instead of some of these we should have V3Width convert LOGIC_IMPLICIT to just logic?   The need for implicit should be only in the front end.

**Comment by yTakatsukasa** [comment]
Thanks for the comment.
I once thought of V3Width too, but not fully sure about the importance of keeping IMPLICIT.
I will try converting IMPLICIT to just logic in V3Width.

**Comment by yTakatsukasa** [comment]
Your advice works!

Making *explicit* the logic dtype in V3Width made the code much simpler as in 80e381ecb3e6277403a7ac180cbdda5d64378a07.
Other code changes are reverted.

**Pull request merged**